functionList/LSL_byKimpaTammas.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- ==========================================================================\
+|
+|   To learn how to make your own language parser, please check the following
+|   link:
+|       https://npp-user-manual.org/docs/function-list/
+|
+\=========================================================================== -->
+<NotepadPlus>
+	<functionList>
+		<!-- ================================================================= [ LSL ] -->
+		<!-- LSL - Linden Script Language                                              -->
+
+		<parser
+			id="lsl"
+			displayName="LSL"
+			commentExpr="(?:(/\*.*?\*/)|(//.*?$))"
+		>
+			<classRange
+				mainExpr="(?x)(?:\b(?:(state)\s+(?:[A-Za-z_]\w*)|(default))\b)(?=\s*\{)"
+				openSymbole="\{"
+				closeSymbole="\}"
+				displayMode="node"
+			>
+				<className>
+					<nameExpr expr="\b(?:(state)\s+(?:[A-Za-z_]\w*))\b" />
+					<nameExpr expr="\b(default)\b" />

```

Comment:
The `nameEpxr`-nodes work as consecutive filters.
The first `expr`-pattern is used to find a match in any text that matched the `mainExpr`.
The second `expr`-pattern is used to find a match in the matched text of the first expression.